### DISCUSSION TITLE
Character limit of `--dotenv` in `run-fargate`

### DISCUSSION THREAD

**Discussion by jhengy** [artilleryio/artillery#2518] [Artillery]
Hi all, 

I was following[ the official CLI doc](https://www.artillery.io/docs/reference/cli/run-fargate) in using `--dotenv` to propagate environment variables from host to fargate containers. It was working fine at the beginning. However it seems like this approach does not scale with more variables. When I add a few more environment variables, I encounter the following error preventing artillery from starting the remote fargate containers:
```
invalidparameterexception: container overrides length must be at most 8192
```

Did a cursory search online and it seems that [AWS imposes a limit of 8192 characters](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RunTask.html ) allowed for the override block in the ECS run task config. I suspect that artillery run-fargate uses it under the hood? 

May I ask for advice on how to overcome this issue? Or is there any workaround for this? Thanks in advance 

p.s. A bit more context, in our use case, we need to propagate environment variables such as access token with hundreds of characters in length.

**Comment by bernardobridge** [comment]
Hi @jhengy 👋 

If you have too many env variables or they're too long, then your best bet would be to store this as a secret in [SSM Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html), in each region you need to run your tests from. Make sure to create any secret with the following format: `/artilleryio/<SECRET_NAME>`.

Then, you can either:
-  use the `--secret` flag (passing in the `SECRET_NAME`'s) - they'll be available as environment variables in workers, same as overriddes.
- or you can use a [`before` hook](https://www.artillery.io/docs/reference/test-script#before-and-after-sections) in your tests to fetch the secrets from SSM programmatically - you can set them in `context.vars` to be available to your test script to use.

**Comment by hassy** [comment]
Created an issue here: https://github.com/artilleryio/artillery/issues/2584 

@jhengy we do indeed rely on task overrides to pass in environment variables to Fargate tasks at the moment. Treating this as a bug as the implementation detail is leaking to the user.
